iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -331,6 +331,23 @@ static void splitParallelAndReductionTiles(
   }
 }
 
+static void setAlwaysVectorizeSizes(linalg::LinalgOp op,
+                                    SmallVectorImpl<int64_t> &parallelSizes,
+                                    SmallVectorImpl<int64_t> &reductionSizes) {
+  Optional<SmallVector<int64_t, 4>> staticLoopRanges = op.getStaticLoopRanges();
```

Comment:
Would this crash if the op doesn't have static loop ranges? Maybe we should return gracefully intead?

Reply:
The ShapedType::kDynamic (i.e., `-1`) will be returned if there are dynamic dims. It's very wrong if we can't get ranges. The ranges are inferred based on indexing_maps and operands. It should be an invalid Linalg op if we can't get ranges.

Maybe we should add it to upstream verifier and remove `Optional` from this method.

Reply:
+1 to removing `Optional` from there. Not sure why it was added to begin with. I think I reviewed that change and suggested dropping the `Optional` as well.

Reply:
Okay, I'll update upstream method and send out fixes to IREE

Reply:
You could land this without those changes. If you look at the method today it never returns None